### ISSUE TITLE
MacOS High Sierra doesn't respect the bssid given

### ISSUE THREAD

**Issue by MuhsinFatih** [qpSHiNqp/airport-bssid#7]
issue #6 persists on High Sierra. This is extremely frustrating because my neighbour has an `ac` capable router (with the same ssid as mine, and I cannot change mine) that is faulty and my mac always wants to connect to it instead. I can't use internet at all

**Comment by MuhsinFatih** [comment]
I am curious, is there a chance anyone has a copy of an old airport executable, from an older version of MacOS? From when it had the option to associate with bssid?

**Comment by Moorviper** [comment]
check out an older revision and compile it ???

**Comment by MuhsinFatih** [comment]
check out where? it's not opensource you know. I searched for older builds but couldn't find one

**Comment by Moorviper** [comment]
clone out the repo / checkout the revision you want and build it with xcode ...

**Comment by MuhsinFatih** [comment]
I am not talking about this repo, I am talking about the airport utility that **comes built-in** in macos. If you read the question a teensy bit carefully before answering you would see that I am talking about the MacOS's built-in airport utility as I said: "**from an older version of MacOS**", not "from an earlier version of this repository" ...

**Comment by Moorviper** [comment]
what has this repo to do with the official airport app.
then this is not a bug with airport-bssid.

It makes no sense  to add a issue here.

**Comment by MuhsinFatih** [comment]
@Moorviper the intent of this issue was to remind that the tool does not work on macos sierra. My comment's intent was to see if I can find potential workaround in the meantime. Are you seriously incapable of interpreting very simple text?? I didn't mention the official airport app **IN THE ISSUE DESCRIPTION** I only mentioned in in comments. Comments are NOT the description of the issue, they are merely comments, hence the name.. Just read the goddamn issue and the comment again instead of proofreading and whining about it. It will take you 10 more seconds ffs.

**Comment by Moorviper** [comment]
ok but airport-bssid works for me on 10.11, 10.12 and 10.13 very well.
Only with networks with band or client steering. But there it couldn't work.

**Comment by alex-laties** [comment]
`airport-bssid` does definitely work on 10.13 High Sierra. The trick though is that the results from `option-click` on the WiFi icon are cached and incorrect after running `airport-bssid` and connecting to a new BSSID.

To check that you have connected to the desired BSSID, you can do the following:
```
~> cd /System/Library/PrivateFrameworks/Apple80211.framework/Versions/A/Resources/
~> ./airport -I | grep BSSID
```

**Comment by MuhsinFatih** [comment]
Can you confirm from router panel as well? I spent days on this issue in my previous apartment and couldn't connect the router that I wanted, could airport cli be reporting wrong bssid?

**Comment by pjvandehaar** [comment]
@alex-laties 
That's not working for me on High Sierra 10.13.6.  Here's what I'm trying:

```
$ git clone https://github.com/qpSHiNqp/airport-bssid.git
$ alias airport=/System/Library/PrivateFrameworks/Apple80211.framework/Versions/Current/Resources/airport

$ sudo airport -z

$ sudo ./airport-bssid/Build/airport-bssid en0 5e:<censored>
Notice: The interface en0 is in none phyMode.
<snipped, but desired network is highlighted>
Associated to network "xfinitywifi" (BSSID: 5e:<censored>)

$ sudo airport -I | grep BSSID
BSSID: e6:<censored>
```

Do you have any advice on what to do differently?

Related to what @Moorviper mentioned, perhaps xfinity routers are able to move me from 5GHz on the working router to 5GHz on the closer, broken one?

**Comment by appleadams** [comment]
I can confirm that I am having the same issue on Mojave.  If their is anything that I can do I'm open to helping.  Also, yah my issue is 100% with xfinitywifi

**Comment by Moorviper** [comment]
yes the newer xfinity routers do band steering by default so it could not work on your networks.

Maybe you can turn it off.

**Comment by frakman1** [comment]
> I am curious, is there a chance anyone has a copy of an old airport executable, from an older version of MacOS? From when it had the option to associate with bssid?

@MuhsinFatih What is the most recent version of OSX that would have this? Maybe I can get it off there or a VM

**Comment by MuhsinFatih** [comment]
@frakman1 I'm sorry, this was a long time ago and I don't remember the research I had done on this, and I'm occupied with other things at the moment
